scripts/post.js

```diff
@@ -332,6 +332,105 @@ function decorateImages() {
   })
 }
 
+/**
+ * Checks if a given match intersects with an existing match
+ * before adding it to the list of matches. In case of an 
+ * intersection, the more specific (i.e. longer) match wins.
+ * @param {array} matches The existing matches
+ * @param {object} contender The match to check and add
+ */
+function checkAndAddMatch(matches, contender) {
+  const collisions = matches
+    // check for intersections
+    .filter((match) => {
+      if (contender.end < match.start || contender.start > match.end) {
+        // no intersection with existing match
+        return false;
+      }
+      // contender starts or ends within existing match
+      return true;
+    })
+    // check for specificity
+    .filter((match) => {
+      if (contender.item.Keyword.length > match.item.Keyword.length) {
```

Comment:
shouldn't the test for better match be before the overlap rejection?

otherwise you have:

text: 'The quick brown fox jumps over the lazy dog.'
kw1: 'fox'
text: 'The quick brown **fox** jumps over the lazy dog.'
kw2: 'brown fox' -> overlaps with kw1 -> rejected

I think if you sort the keywords by length descending, you don't need this check.

Reply:
(a unit test, would allow to quickly test all edge cases :-)

Reply:
I couldn't reproduce the `fox` keyword winning over the `brown fox`, but you were right with sorting keywords by length before matching: this produces better results with less code 🙂